src/main/java/com/arpnetworking/tsdcore/sinks/AggregationServerSink.java

```diff
@@ -50,25 +50,25 @@ public void recordAggregateData(final PeriodicData periodicData) {
                 .addData("dataSize", periodicData.getData().size())
                 .addData("conditionsSize", periodicData.getConditions().size())
                 .log();
-
-        final Map<String, List<AggregatedData>> data = periodicData.getData()
-                .stream()
-                .filter(datum -> !EXPRESSION_STATISTIC.equals(datum.getFQDSN().getStatistic()))
-                .collect(Collectors.groupingBy(d -> d.getFQDSN().getMetric()));
-        for (final List<AggregatedData> dataList : data.values()) {
-            if (!dataList.isEmpty()) {
-                final Messages.StatisticSetRecord record = serializeMetricData(periodicData, dataList);
+        
+        for (final Map.Entry<String, Collection<AggregatedData>> entry : periodicData.getData().asMap().entrySet()) {
+            final String metricName = entry.getKey();
+            final Collection<AggregatedData> data = entry.getValue();
+            if (!data.isEmpty()) {
+                final Messages.StatisticSetRecord record = serializeMetricData(periodicData, metricName, data);
                 enqueueData(AggregationMessage.create(record).serialize());
             }
         }
     }
 
-    private Messages.StatisticSetRecord serializeMetricData(final PeriodicData periodicData, final List<AggregatedData> dataList) {
-        // Get the cluster from the first data element
-        final AggregatedData firstElement = dataList.iterator().next();
+    private Messages.StatisticSetRecord serializeMetricData(
+            final PeriodicData periodicData,
+            final String metricName,
+            final Collection<AggregatedData> data) {
 
+        // Map all dimensions
         final List<Messages.DimensionEntry> dimensions = Lists.newArrayList();
-        for (final Map.Entry<String, String> entry : periodicData.getDimensions().entrySet()) {
+        for (final Map.Entry<String, String> entry : periodicData.getDimensions().getDimensions().entrySet()) {
```

Comment:
I just renamed the method in Set from _getDimensions_ to _getParameter_. Does that work for you?